astropy/time/formats.py

```diff
@@ -2175,6 +2176,204 @@ def value(self):
+        # with the default values.
+        components = ("yr", "d", "hr", "min", "s")
+
+        if (match := self.re_ydhms.match(timestr)) is None:
+            raise ValueError(
+                f"Time delta '{timestr}' does not match {self.name} format"
+            )
+
+        tm = match.groupdict()
+        vals = [float(tm[component] or 0.0) for component in components]
+        if tm["sign"] == "-":
+            vals = [-val for val in vals]
+
+        return vals
+
+    def set_jds(self, val1, val2):
+        """Parse the time strings contained in val1 and get jd1, jd2."""
+        # Be liberal in what we accept: convert bytes to ascii.
+        # Here .item() is needed for arrays with entries of unequal length,
+        # to strip trailing 0 bytes.
+        to_string = (
+            str if val1.dtype.kind == "U" else lambda x: str(x.item(), encoding="ascii")
+        )
+        iterator = np.nditer(
+            [val1, None, None, None, None, None],
+            flags=["zerosize_ok"],
+            op_dtypes=[None] + 5 * [np.double],
+        )
+        for val, yr, day, hr, min, sec in iterator:
+            val = to_string(val)
+            (
+                yr[...],
+                day[...],
+                hr[...],
+                min[...],
+                sec[...],
+            ) = self.parse_string(val)
+
+        yrs, days, hrs, mins, secs = iterator.operands[1:]
+        jd1 = yrs * 365.25 + days + hrs / 24.0 + mins / 1440.0 + secs / 86400.0
+        self.jd1, self.jd2 = day_frac(jd1, 0.0)
+
+    def to_value(self, parent=None, out_subfmt=None):
+        out_subfmt = out_subfmt or self.out_subfmt
+        subfmt = self._get_allowed_subfmt(out_subfmt)
+
+        iterator = np.nditer(
+            [self.jd1, self.jd2, None],
+            flags=["refs_ok", "zerosize_ok"],
+            op_dtypes=[None, None, object],
+        )
+
+        for jd1, jd2, out in iterator:
+            jd1_, jd2_ = day_frac(jd1, jd2)  # Why is this required?
+
+            # Naive and low-precision implementation for now
+            jd = jd1_ + jd2_
+            sign = "-" if jd < 0 else ""
+            jd = np.abs(jd)
+
+            if subfmt in ["*", "multi"]:
+                comps = self.get_multi_comps(jd)
+            else:
+                value = (jd * u.day).to_value(subfmt)
+                value = np.round(value, self.precision)
+                comps = [f"{value}{subfmt}"]
+
+            out[...] = sign + " ".join(comps)
+
+        return self.mask_if_needed(np.array(iterator.operands[-1], dtype="U"))
+
+    def get_multi_comps(self, jd):
+        comp_names = ("yr", "d", "hr", "min", "s")
+        comp_scales = (365.25, 1.0, 1.0 / 24.0, 1.0 / 1440.0)
+        comps = []
+        for name, scale in zip(comp_names, comp_scales):
+            comp, jd = divmod(jd, scale)
+            if comp != 0:
+                comp = int(comp)
+                comps.append(f"{comp}{name}")
+
+        sec = np.round(jd * 86400.0, self.precision)
```

Comment:
Not directly related to the larger question, but this rounding at the end means you can wind up with `60.0s` - do you want to special-case that? As is,
```
TimeDelta(365.9999999999, format='jd').quantity_str
# '1yr 17hr 59min 60.0s'
```

To avoid duplication, maybe consider using some of the code in `astropy/coordinates/angles/formats.py`; as is,
```
from astropy.coordinates.angles import formats
formats.hours_to_string(.99999999999*24, precision=3, sep=['hr ', 'min ', 's'])
# '24hr 00min 00.000s'
```
I think the rounding bit could be quite easily factored out of `sexagesimal_to_string`.

Reply:
I was thinking about that corner case this morning and agree this should be done right. Thanks for the heads-up on existing code, I'll have a look.

Reply:
Fixed